File: 222-CountCompleteTreeNodes/222-CountCompleteTreeNodes.java
// Last updated: 01/09/2025, 01:04:57
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int cnt = 0;
    public int countNodes(TreeNode root) {
        if(root == null) return 0;
        countNodes(root.left);
        countNodes(root.right);
        cnt += 1;
        return cnt;
    }
}

File: 222-CountCompleteTreeNodes/222-CountCompleteTreeNodes.cpp
// Last updated: 01/09/2025, 01:05:02
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int cnt = 0;
    int countNodes(TreeNode* root) {
        if(root == NULL) return 0;
        countNodes(root -> left);
        countNodes(root -> right);
        cnt++;
        return cnt;
    }
};